pkg/cloud/vsphere/context/cluster_context.go

```diff
@@ -344,7 +341,10 @@ func (c *ClusterContext) Patch() error {
 		c.Cluster.ResourceVersion = result.ResourceVersion
 	}
 
-	c.Cluster.Status.ProviderStatus = newStatus
+	// Put the status back.
+	c.Cluster.Status = clusterv1.ClusterStatus{}
```

Comment:
I was expecting to see `result` on line 332 get its status updated (and if there was no patch, then you could use the existing object) and have that saved.

But if it works like this then that's 💯

Reply:
Hi @chuckha,

Hmm, we just followed the CAPA model at https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/ace9cb7b1ab5b3d5ced5663e44656ac94ad78b2c/pkg/cloud/aws/actuators/scope.go#L204. Maybe there was a reason for it?